### PR TITLE
feat: add support for disabling IGW for public subnets

### LINKED ISSUE #446: feature request: allow disabling generation of igw
Use cases:

1) Internet traffic is routed via an Egress VPC via a transit gateway (for outbound content filtering), but there are still the distinction of a "public" and a "private" network inside the VPC.
2) Alternatively the traffic for the "public" network is routed via a DirectConnect (great way of bringing publicly routed IPs into your VPC onto the EC2 hosts without NAT involved).

Solution:
Leave the default behaviour as is but add an additional check to the igw creation to also check for `disable_igw_creation` (default of `false`).

### PR DESCRIPTION
**Pull request by DrFaust92**
## Description

Add support for disabling of creating an IGW for public subnets for a case that is specified in #446

## Motivation and Context

Closes #446

## Breaking Changes

No

## How Has This Been Tested?

Ran simple example with the `disable_igw_creation` set to `true`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -325,6 +325,7 @@ It is possible to integrate this VPC module with [terraform-aws-transit-gateway
 | dhcp\_options\_netbios\_node\_type | Specify netbios node\_type for DHCP options set (requires enable\_dhcp\_options set to true) | `string` | `""` | no |
 | dhcp\_options\_ntp\_servers | Specify a list of NTP servers for DHCP options set (requires enable\_dhcp\_options set to true) | `list(string)` | `[]` | no |
 | dhcp\_options\_tags | Additional tags for the DHCP option set (requires enable\_dhcp\_options set to true) | `map(string)` | `{}` | no |
+| disable\_igw\_creation | Controls if an Internet Gateway is created for public subnets and the related routes that connect them. | `bool` | `false` | no |
```
Comment (antonbabenko) [inline README.md:328]: Let's change the name to the opposite which matches the rest of conditional arguments - `create_igw`.

Comment (DrFaust92) [reply README.md]: done, should this be added for the egress IGW as well?

Comment (antonbabenko) [reply README.md]: yes, probably. It is a good idea to control everything in general.

Comment (DrFaust92) [reply README.md]: added a different flag for egress igw and related routes to give a more granular control

**Comment by antonbabenko** [comment]
[v2.43.0](https://github.com/terraform-aws-modules/terraform-aws-vpc/releases/v2.43.0) has been just released.

**Pull request merged**